Return NotFound error as-is in GetGlobalNetworks

vthapar opened the pull request:
In GetGlobalNetworks we wrap error returned with a custom error.
This breaks the validation check in calling methods if they want
to ignore IsNotFound error. Fix is to return IsNotFound error as is.

Signed-off-by: Vishal Thapar <5137689+vthapar@users.noreply.github.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

vthapar committed:
Return NotFound error as-is in GetGlobalNetworks

In GetGlobalNetworks we wrap error returned with a custom error.
This breaks the validation check in calling methods if they want
to ignore IsNotFound error. Fix is to return IsNotFound error as is.

Signed-off-by: Vishal Thapar <5137689+vthapar@users.noreply.github.com>

sridhargaddam commented on pkg/discovery/globalnet/globalnet.go:
```diff
@@ -293,6 +293,9 @@ func ValidateGlobalnetConfiguration(globalnetInfo *GlobalnetInfo, netconfig Conf
 func GetGlobalNetworks(k8sClientset *kubernetes.Clientset, brokerNamespace string) (*GlobalnetInfo, *v1.ConfigMap, error) {
 	configMap, err := broker.GetGlobalnetConfigMap(k8sClientset, brokerNamespace)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, nil, err
+		}
 		return nil, nil, fmt.Errorf("error reading configMap: %s", err)
```
```suggestion
		return nil, nil, err
```

I feel we can just return err and let the caller check for the type of error.

mangelajo replied:
you can also do errors.Wrap(err, "error reading configMap") which I think will preserve all the other attributes (being able to call IsNotFound, etc).

Always better to use errors.Wrapf / errors.Wrap

skitt replied:
Yes, always use `errors.Wrap()`. Technically, this doesn’t preserve the original error’s _attributes_ directly, but it supports `Unwrap()`, which is [the documented way of handling wrapped errors](https://pkg.go.dev/errors) and [is supported by K8s’ `.Is...` functions](https://pkg.go.dev/k8s.io/apimachinery/pkg/api/errors).

skitt committed:
Ensure we use our new subctl

We need to test the new subctl during deployment; our specific PATH
setting was lost with dc201779cb47 ("Remove Dockerfile.dapper"), this
restores it in the Makefile and ensures bin/subctl is built before
deployment.

Signed-off-by: Stephen Kitt <skitt@redhat.com>
